### PR TITLE
update boost to 1.78.0

### LINKED ISSUE #2550: [Update request]: update boost
1.60 is from Dec 2015. Would be nice to get a more recent version included.

Related: #2302 #2043 #1549

### PR DESCRIPTION
**Pull request by orgads**
Fixes #2550.

### CODE REVIEW

**Comment by orgads** [comment]
ping?

**Comment by orgads** [comment]
@mabrand ping

File: src/boost.mk [hunk 1]
```diff
@@ -69,14 +71,14 @@ define $(PKG)_BUILD
...
-        -lboost_serialization-mt \
-        -lboost_thread_win32-mt \
-        -lboost_system-mt \
-        -lboost_chrono-mt \
-        -lboost_context-mt
+        -lboost_serialization$($(PKG)_SUFFIX) \
+        -lboost_thread$($(PKG)_SUFFIX) \
+        -lboost_system$($(PKG)_SUFFIX) \
+        -lboost_chrono$($(PKG)_SUFFIX) \
+        -lboost_context$($(PKG)_SUFFIX)
```
Comment (mabrand) [inline src/boost.mk:81]: Could you explain this change? I don't think I understand where $(PKG)_SUFFIX is set.

File: src/boost.mk [hunk 2]
```diff
@@ -14,6 +14,8 @@ $(PKG)_DEPS     := cc bzip2 expat zlib
 
 $(PKG)_DEPS_$(BUILD) := zlib
 
+$(PKG)_SUFFIX = -mt-x$(if $(findstring x86_64,$(TARGET)),64,32)
```
Comment (orgads) [inline src/boost.mk:17]: Here

File: src/boost.mk (hunk 1 above)
Comment (orgads) [reply src/boost.mk:81]: The library names were changed from just `-mt` to `-mt-x32` or `-mt-x64`, depends on the target platform.

**Comment by orgads** [comment]
@mabrand Can you please create a new tag following this merge?

**Comment by mabrand** [comment]
tag build-2022-01-13?

**Comment by orgads** [comment]
Sure, thanks.

**Comment by mabrand** [comment]
done

**Pull request merged**